script/src/types.rs

```diff
@@ -92,6 +92,14 @@ impl ScriptVersion {
         let version = self.vm_version();
         CoreMachineType::new(isa, version, max_cycles)
     }
+
+    #[cfg(test)]
+    pub(crate) fn use_dirty_flag_when_load_data_as_code(self) -> bool {
```

Comment:
This is only for test.

We should run each tests in 3 scenes, but current test framework only supports 2.
- (Tested) `vm0`, `dirty-flag=false`
- (Skipped) `vm0`, `dirty-flag=true`
- (Tested)`vm1`, `dirty-flag=true`

This method use the same logic as the follow code (no `vm0` test after hardfork):
https://github.com/nervosnetwork/ckb/blob/3f405a0f5dc4ccbfb294d202ae9f4396691ca7b5/script/src/verify/tests/utils.rs#L152-L153